### PR TITLE
add cancel order option

### PR DESCRIPTION
**Pull request by Victoradukwu**
#### What does this PR do?
- Add cancel order functionality
#### Description of Task to be completed?
- Add a cancel order button that cancels a pending order
- When the order is cancelled, the amount is refunded into the customer’s wallet.
- The customer receives a notification of the cancelled order
#### How should this be manually tested?
- Start the app with the command `reaction`
- Go to the port 3000
- Add a product to your cart and checkout with the wallet. A cancel order button appears at your profile
- Click the button to cancel the order
- Confirm cancel order
- An alert will show the cancellation and an in-app notification all be sent.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
 - Add `Cancel Order` option

#### Screenshots (if appropriate)
image
image
image
image
image
image

#### Questions:
N/A

### CODE REVIEW

File: imports/plugins/core/checkout/client/components/completedOrder.js [hunk 1]
```diff
@@ -18,95 +20,219 @@ import AddEmail from "./addEmail";
...
+           (err, res) => {
+             if (res === 2) {
+               Alerts.toast(`No user with email ${email}`, "error");
+             } else if (res === 1) {
+               Alerts.toast("Order successfully cancelled", "success");
+               this.setState({
+                 isCancelled: true
+               });
+             } else {
+               Alerts.toast("An error occured, please try againnnnnn", "error");
```
Comment (michaelogala) [inline imports/plugins/core/checkout/client/components/completedOrder.js]: One `n` is enough in `again`

File: imports/plugins/core/checkout/client/components/completedOrder.js [hunk 2]
```diff
@@ -18,95 +20,219 @@ import AddEmail from "./addEmail";
...
+           amount: Number(amount),
+           to: email,
+           from: email,
+           transactionType: "credit"
+         };
+         Meteor.call("orders/cancelOrder", order, true);
+         Meteor.call("wallet/update-balance",
+           transaction,
+           (err, res) => {
+             if (res === 2) {
```
Comment (michaelogala) [inline imports/plugins/core/checkout/client/components/completedOrder.js]: What does 2 and 1 mean here and on line 61?

File: imports/plugins/core/ui-navbar/client/components/navbar.js [hunk 3]
```diff
@@ -61,13 +61,13 @@ class NavBar extends Component {
   }
 
   renderBrand() {
-    const shop = this.props.shop || { name: "" };
+    // const shop = this.props.shop || { name: "" };
```
Comment (michaelogala) [inline imports/plugins/core/ui-navbar/client/components/navbar.js]: Remove commented code

**Review by michaelogala — CHANGES_REQUESTED** [review]

File: imports/plugins/core/checkout/client/components/completedOrder.js (hunk 2 above)
Comment (Victoradukwu) [reply imports/plugins/core/checkout/client/components/completedOrder.js]: That figure `2` is actually supposed to be `0`. The **wallet/update-balance** method returns 1 for success and 0 for error. I have applied the changes accordingly.
I have also corrected the typo and removed the commented codes.

File: imports/plugins/included/wallet/server/methods/wallet.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    } = transaction;
+
+    const updateBalance = (currentBalance, updateAmount, ownerEmail) => {
+      try {
+        Wallets.update({ ownerEmail }, {
+          $set: {
+            balance: currentBalance + updateAmount
+          }
+        });
+        return 1;
```
Comment (michaelogala) [inline imports/plugins/included/wallet/server/methods/wallet.js]: Please change this to `return 'Success'`. Also change line 161 to `return 'Error'`. Let this change reflect in lines 59 and 61 of `completeOrder.js`

**Review by michaelogala — CHANGES_REQUESTED** [review]

**Comment by michaelogala** [comment]
@Victoradukwu please resolve the merge conflicts and make the changes requested

File: imports/plugins/included/wallet/server/methods/wallet.js (hunk 4 above)
Comment (Victoradukwu) [reply imports/plugins/included/wallet/server/methods/wallet.js]: I have implemented the requested changes.

**Review by michaelogala — APPROVED** [review]

**Pull request merged**